depends/funcs.mk

```diff
@@ -157,48 +158,60 @@ $(1)_autoconf += LDFLAGS="$$($(1)_ldflags)"
 endif
 endef
 
+shortdate="+%H:%M:%S"
+
 define int_add_cmds
-$($(1)_fetched):
+$($(1)_introduced):
+	$(AT)echo "============================================================"
+	$(AT)echo "   Package : $(1)"
+	$(AT)echo "   Date    : `date`"
+	$(AT)echo "============================================================"
+	$(AT)mkdir -p $$(@D)
+	$(AT)touch $$@
+$($(1)_fetched): | $($(1)_introduced)
 	$(AT)mkdir -p $$(@D) $(SOURCES_PATH)
 	$(AT)rm -f $$@
 	$(AT)touch $$@
-	$(AT)cd $$(@D); $(call $(1)_fetch_cmds,$(1))
+	$(AT)cd $$(@D); ( echo -n -e ""; $(call $(1)_fetch_cmds,$(1)) ) > .fetch_output 2>&1 || ( cat .fetch_output && false )
 	$(AT)cd $($(1)_source_dir); $(foreach source,$($(1)_all_sources),$(build_SHA256SUM) $(source) >> $$(@);)
 	$(AT)touch $$@
-$($(1)_extracted): | $($(1)_fetched)
-	$(AT)echo Extracting $(1)...
+$($(1)_extracted): | $($(1)_introduced) $($(1)_fetched)
+	$(AT)echo `date $(shortdate)`: Extracting $(1)...
 	$(AT)mkdir -p $$(@D)
-	$(AT)cd $$(@D); $(call $(1)_extract_cmds,$(1))
+	$(AT)cd $$(@D); ( echo -n -e ""; $(call $(1)_extract_cmds,$(1)) ) > .extract_output 2>&1 || ( cat .extract_output && false )
 	$(AT)touch $$@
-$($(1)_preprocessed): | $($(1)_dependencies) $($(1)_extracted)
-	$(AT)echo Preprocessing $(1)...
+$($(1)_preprocessed): | $($(1)_introduced) $($(1)_dependencies) $($(1)_extracted)
+	$(AT)echo `date $(shortdate)`: Preprocessing $(1)...
 	$(AT)mkdir -p $$(@D) $($(1)_patch_dir)
 	$(AT)$(foreach patch,$($(1)_patches),cd $(PATCHES_PATH)/$(1); cp $(patch) $($(1)_patch_dir) ;)
-	$(AT)cd $$(@D); $(call $(1)_preprocess_cmds, $(1))
+	$(AT)cd $$(@D); ( echo -n -e ""; $(call $(1)_preprocess_cmds, $(1)) ) > .preproc_output 2>&1 || ( cat .preproc_output && false )
 	$(AT)touch $$@
-$($(1)_configured): | $($(1)_preprocessed)
-	$(AT)echo Configuring $(1)...
+$($(1)_configured): | $($(1)_introduced) $($(1)_preprocessed)
+	$(AT)echo `date $(shortdate)`: Configuring $(1)...
 	$(AT)rm -rf $(host_prefix); mkdir -p $(host_prefix)/lib; cd $(host_prefix); $(foreach package,$($(1)_all_dependencies), tar xf $($(package)_cached); )
 	$(AT)mkdir -p $$(@D)
-	$(AT)+cd $$(@D); $($(1)_config_env) $(call $(1)_config_cmds, $(1))
+	$(AT)+cd $$(@D); ( echo -n -e ""; $($(1)_config_env) $(call $(1)_config_cmds, $(1)) ) > .config_output 2>&1 || ( cat .config_output && false )
 	$(AT)touch $$@
 $($(1)_built): | $($(1)_configured)
-	$(AT)echo Building $(1)...
+	$(AT)echo `date $(shortdate)`: Building $(1)...
+	$(AT)rm -f .built-$(1)
+	$(AT)( sleep 540; if [ ! -e ".built-$(1)" ]; then echo `date $(shortdate)`: ...; sleep 540; if [ ! -e ".built-$(1)" ]; then echo `date $(shortdate)`: ...; fi; fi ) &
```

Comment:
I don't understand what this is supposed to be doing. Looks broken, though :)

Reply:
It prints out "..." after 9 and 18 minutes, unless ".built-$(1)" has been deleted. It prevents travis from killing the qt build, which takes 17 mins.

Reply:
Ah, I missed the "&" at the end.

I don't think we want to be concerned about that at this level. IMO we should just use ```travis_wait=20``` or so for making depends.

Reply:
I initially went that route, but travis_wait seems to block output completely.